lib/mix/lib/mix/scm/hg.ex

```diff
@@ -0,0 +1,172 @@
+defmodule Mix.SCM.Mercurial do
+  @behaviour Mix.SCM
+  @moduledoc false
+  
+  def fetchable? do
+    true
+  end
+  
+  def format(opts) do
+    opts[:hg]
+  end
+  
+  def format_lock(opts) do
+    case opts[:lock] do
+      {:hg, _, lock_rev, lock_opts} ->
+        lock = String.slice(lock_rev, 0, 7)
+        case Enum.find_value [:branch, :ref, :tag], &List.keyfind(lock_opts, &1, 0) do
+          {:ref, _}  -> lock <> " (ref)"
+          {key, val} -> lock <> " (#{key}: #{val})"
+          nil        -> lock
+        end
+      _ ->
+        nil
+    end
+  end
+  
+  def accepts_options(_app, opts) do
+    cond do
+      repo = opts[:bitbucket] ->
+        opts |> Keyword.delete(:bitbucket) |> Keyword.put(:hg, "https://bitbucket.org/#{repo}")
+      repo = opts[:hg_github] ->
```

Comment:
No problem, I can remove this. My team and I use hg-git to work on Github repos, so I tried to pick an obscure but intuitive keyword that supports this.